backend/src/main/java/ai/verta/modeldb/versioning/CommitDAORdbImpl.java

```diff
@@ -315,16 +285,106 @@ public CreateCommitRequest.Response setCommitFromDatasetVersion(
     }
   }
 
+  @NotNull
+  private Blob getBlobBuilderFromDatasetVersion(DatasetVersion datasetVersion) {
+    var blobBuilder = Blob.newBuilder();
+    if (datasetVersion.hasDatasetBlob()) {
+      blobBuilder.setDataset(populateInternalPathInDatasetBlob(datasetVersion.getDatasetBlob()));
+    } else {
+      var datasetBlobBuilder = DatasetBlob.newBuilder();
+      switch (datasetVersion.getDatasetVersionInfoCase()) {
```

Comment:
with only one valid branch, an if is probably better here.

Reply:
Sorry but I do not understand this comment